api/api.go

```diff
@@ -359,15 +360,33 @@ func (api *API) delSilence(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
-	sils, err := api.silences.All()
+	var (
+		num    = r.FormValue("n")
+		offset = r.FormValue("offset")
```

Comment:
Offset based pagination has some disadvantages. Besides being quite slow with several relational-databases (which doesn't really matter here), it's not as stable as an after-id based approach. Using something like `?after=last-seen-id&limit=50` will ensure that the user will really see the next result when using pagination.